data_safe_haven/commands/config.py

```diff
@@ -22,7 +26,14 @@ def show_shm(
 ) -> None:
     """Print the SHM configuration for the selected Data Safe Haven context"""
     context = ContextManager.from_file().assert_context()
-    config = SHMConfig.from_remote(context)
+    logger = get_logger()
+    try:
+        config = SHMConfig.from_remote(context)
+    except DataSafeHavenError as exc:
+        logger.critical(
+            "SHM must be deployed before its configuration can be displayed."
+        )
+        raise typer.Exit(1) from exc
```

Comment:
It would be good to test the extra "user interface" errors introduced here.

It should be quite simple, for example here mock `from_remote` to do nothing except raise `DataSafeHavenError`.